CHANGELOG.md

```diff
@@ -5,6 +5,32 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [v3.74.0](https://github.com/buildkite/agent/tree/v3.73.1+1) (2024-06-06)
+[Full Changelog](https://github.com/buildkite/agent/compare/v3.73.1...v3.74.0)
+
+### Added
+- kubernetes-exec is now a flag [#2814](https://github.com/buildkite/agent/pull/2814) (@DrJosh9000)
+- Add logs to allowed-[repositories|plugins] [#2810](https://github.com/buildkite/agent/pull/2810) (@jakubm-canva)
+- PTY rows/cols increased [#2806](https://github.com/buildkite/agent/pull/2806) (@pda)
+- Dont sign initial steps with interpolations [#2813](https://github.com/buildkite/agent/pull/2813) (@moskyb)
+
+### Fixed
+- Fix error in k8s after job completes [#2804](https://github.com/buildkite/agent/pull/2804) (@DrJosh9000)
+
+### Changed
+- shell logger: Use fmt functions once [#2805](https://github.com/buildkite/agent/pull/2805) (@DrJosh9000)
+- Update deprecated import [#2811](https://github.com/buildkite/agent/pull/2811) (@DrJosh9000)
+- Use Rand per-test in agent/plugin/error_test.go [#2795](https://github.com/buildkite/agent/pull/2795) (@moskyb)
```

Comment:
These (and the dependabot updates below) are internal, mostly non-user visible code changes (or at least should be), so we've typically put them under an `### Internal` header rather than `### Changed`.